pkg/auth/dependency/identity/loginid/store.go

```diff
@@ -166,6 +166,9 @@ func (s *Store) Create(i *Identity) error {
 
 	_, err = s.SQLExecutor.ExecWith(q)
 	if err != nil {
+		if db.IsUniqueViolated(err) {
```

Comment:
Personally I'd avoid checking for unique constraint violation error, because it is hard to recover from errors in transaction using application code, and it may be unreliable depending on isolation level setting. Is it possible to check for duplicated keys explicitly?

Reply:
Seems the unique checking behaves differently in different identity type. I updated the implementation in e7813d1 and do the unique key checking in login id provider.